docs/cram-tests/index.md

```diff
@@ -0,0 +1,457 @@
+adding this to your `dune-project` file:
+
+```dune
+; Enable cram tests
+(cram enable)
+```
+
+## `cram` stanza
+
+To add dependencies for your cram tests, you need to add a [cram
+stanza](https://dune.readthedocs.io/en/stable/dune-files.html#cram-stanza) to
+`src/order-confirmation/dune`:
+
+```dune
+(cram
+ (deps ./output/src/order-confirmation/SandwichTests.mjs))
+```
+
+This configuration tells Dune two things:
+
+- Copy `./output/src/order-confirmation/SandwichTests.mjs` into the sandbox
+  directory
+- Whenever `SandwichTests.re` (the immediate dependency for
+  `./output/src/order-confirmation/SandwichTests.mjs`) changes, rebuild and
+  re-run all the cram tests defined in the same directory as this `dune` file
+
+Note that we cannot use the same path to `SandwichTests.mjs` as we would if we
+were running this from the root directory of our project, because cram tests are
+run inside a sandbox directory (more on that later).
+
+## Add `tests.t` file
+
+Cram tests are specified inside `.t` files. Add a new file
+`src/order-confirmation/tests.t`:
+
+```cram
+Sandwich tests
+  $ node ./output/src/order-confirmation/SandwichTests.mjs
+```
+
+Note that this is the same path for `SandwichTests.mjs` that we specified inside
+the `cram/deps` field.
+
+To run tests, execute
+
+```shell
+dune build @runtest
+```
+
+Here,
+[@runtest](https://dune.readthedocs.io/en/stable/reference/aliases.html#runtest)
+is a built-in *alias*. The above command builds whatever is needed by the tests
+and then runs the tests.
+
+## Aliases
+
+A Dune [alias](https://dune.readthedocs.io/en/stable/reference/aliases.html) is
+a virtual build target that other build targets can attach to. You can refer to
+aliases by name in build commands (like the one above), to build a specific
+subset of your build targets. In commands, we need to put `@` in front of the
+alias name to distinguish it from file names. Therefore when we refer to an
+alias, we put an `@` in front, e.g. `@runtest`, `@melange`, etc.
+
+::: tip
+
+Cram tests are automatically attached to the `@runtest` alias, and
+`melange.emit` stanzas are automatically attached to the `@melange` alias.
+
+:::
+
+## Promotion
+
+After running `dune build @runtest`, you should be able to see the
+output from the cram test you just added[^1]:
+
+<<< initial-diff{diff}
+
+However, this doesn't count as a passing cram test, because this output can't be
+compared to any expected output. Dune has a nice feature that lets you
+[promote](https://dune.readthedocs.io/en/stable/concepts/promotion.html#promotion)
+the latest output of a cram test to become the expected output:
+
+```shell
+dune promote
+```
+
+The above command updates your `src/order-confirmation/tests.t` cram test file,
+which now looks something like this:
+
+<<< tests#first-cram{cram}
+
+The part underneath the command is the expected output of the command.
+
+## Add `npm` scripts
+
+Instead of typing `dune` commands over and over, let's add the relevant `npm`
```

Comment:
Now that `opam exec` is gone this makes less sense 😄 because it's shorter to run `dune test` than `npm run test`. But maybe folks prefer npm, I guess it doesn't hurt.

Reply:
Yeah, it's not really necessary to add the "test" script, just for the sake of completeness I guess